src/core/client/stream/tabs/Live/helpers/handleNewCommentInStory.ts

```diff
@@ -7,31 +7,33 @@ import {
 import insertComment from "./insertComment";
 import markHasNextPage from "./markHasNextPage";
 
-export default function insertCommentToStory(
+export default function handleNewCommentInStory(
   store: RecordSourceSelectorProxy<unknown>,
   storyID: string,
   comment: RecordProxy,
   options: {
     liveInsertion?: boolean;
-    fromMutation?: boolean;
   } = {}
 ) {
   const story = store.get(storyID)!;
   const afterConnection = ConnectionHandler.getConnection(story, "Chat_after");
   if (!afterConnection) {
     return;
   }
+  const pageInfoAfter = afterConnection.getLinkedRecord("pageInfo")!;
+  const hasNextPage = pageInfoAfter.getValue("hasNextPage")!;
+
   const beforeConnection = ConnectionHandler.getConnection(
     story,
     "Chat_before"
   );
   if (options.liveInsertion) {
-    insertComment(store, comment, afterConnection);
-  } else {
-    markHasNextPage(
-      afterConnection,
-      beforeConnection,
-      Boolean(options.fromMutation)
-    );
+    if (!hasNextPage) {
+      insertComment(store, comment, afterConnection);
+    }
+    return;
+  }
+  if (!hasNextPage) {
+    markHasNextPage(afterConnection, beforeConnection);
```

Comment:
Ooof, that `return` complicates things a bit, eh? I started to try and write a cleaner version of this, only to realize I'd end up with 3 flat if's instead of a clean `if...else if`...

I might have done this, that way you can comment each specific flow with flat if's:

```
// live insertion, with no next page, we can simply insert the 
// comment and move along
if (options.liveInsertion && !hasNextPage) {
   insertComment(store, comment, afterConnection);
   return;
}
// not live inserting, not tailing, hasNextPage is true, we
// will just let the existing pagination flag handle it
else if (options.liveInsertion && hasNextPage) {
   return;
}

// not tailing, not live inserting, so we can just mark the
// pagination to handle this
if (!hasNextPage) {
   markHasNextPage(afterConnection, beforeConnection);
}
```